core/templates/dev/head/pages/exploration-editor-page/editor-navigation/editor-navigation.directive.ts

```diff
@@ -43,15 +47,30 @@ angular.module('oppia').directive('editorNavigation', [
       controller: [
         '$rootScope', '$scope', '$timeout', '$uibModal', 'ContextService',
         'ExplorationFeaturesService', 'ExplorationRightsService',
-        'ExplorationWarningsService', 'RouterService',
+        'ExplorationWarningsService', 'ImprovementTaskService',
+        'ImprovementsDisplayService', 'RouterService',
         'SiteAnalyticsService', 'StateTutorialFirstTimeService',
         'ThreadDataService', 'UserService', 'WindowDimensionsService',
         function(
             $rootScope, $scope, $timeout, $uibModal, ContextService,
             ExplorationFeaturesService, ExplorationRightsService,
-            ExplorationWarningsService, RouterService,
+            ExplorationWarningsService, ImprovementTaskService,
+            ImprovementsDisplayService, RouterService,
             SiteAnalyticsService, StateTutorialFirstTimeService,
             ThreadDataService, UserService, WindowDimensionsService) {
+          var fetchedTasks = [];
+          ImprovementTaskService.fetchTasks().then(function(tasks) {
+            fetchedTasks = tasks;
+          });
+
+          $scope.isTaskOpen = function(task) {
+            return ImprovementsDisplayService.isOpen(task.getStatus());
+          };
+
+          $scope.getOpenTaskCount = function() {
```

Comment:
should we create a variable instead? I mean is this going to change?

(I'm fine with this as well.)

Reply:
When anyone send a feedback or a suggestion it should change.

Reply:
This should be a part of the ImprovementTaskService, let's add a method there for "getOpenTaskCount" so it can handle the calculation more efficiently.